src/render/SpecialRenderer.ts

```diff
@@ -670,7 +698,35 @@ export namespace SpecialRenderers {
 						down: {uv: [10.5, 10.5, 5.5, 11], texture: '#0'},
 					},
 				},
-			]).getMesh(atlas, Cull.none())
+			]
+			const colors: string[] = [color]
+
+			if (nbt !== undefined) {
+				const patterns = nbt.getList('patterns', 10)
+				patterns.forEach((compound, index) => {
+					const pattern = Identifier.parse(compound.getString('pattern')).path
+					const color = compound.getString('color')
+					const key = index + 1
+					textures[key] = `entity/banner/${pattern}`
+					elements.push({
+						from: [-2, -8, -1.5],
+						to: [18, 32, -0.5],
+						faces: {
+							north: {uv: [0.25, 0.25, 5.25, 10.25], texture: `#${key}`, tintindex: key},
+							east: {uv: [0, 0.25, 0.25, 10.25], texture: `#${key}`, tintindex: key},
+							south: {uv: [5.5, 0.25, 10.5, 10.25], texture: `#${key}`, tintindex: key},
+							west: {uv: [5.25, 0.25, 5.5, 10.25], texture: `#${key}`, tintindex: key},
+							up: {uv: [5.25, 0.25, 0.25, 0], texture: `#${key}`, tintindex: key},
+							down: {uv: [10.25, 0, 5.25, 0.25], texture: `#${key}`, tintindex: key},
+						},
+					})
+					colors.push(color)
+				})
+			}
```

Comment:
This seems like a lot of duplicated code. Could this be moved into its own function to be used by both banner renderers?